packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -417,45 +428,46 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       }
     };
     // default model operations
-    addServiceDirective(ctx.output.getQueryTypeName()!, 'read', modelConfig?.queries?.get);
-    addServiceDirective(ctx.output.getQueryTypeName()!, 'read', modelConfig?.queries?.list);
-    addServiceDirective(ctx.output.getQueryTypeName()!, 'read', modelConfig?.queries?.sync);
+    addServiceDirective(ctx.output.getQueryTypeName()!, 'get', modelConfig?.queries?.get);
+    addServiceDirective(ctx.output.getQueryTypeName()!, 'list', modelConfig?.queries?.list);
+    addServiceDirective(ctx.output.getQueryTypeName()!, 'sync', modelConfig?.queries?.sync);
     addServiceDirective(ctx.output.getMutationTypeName()!, 'create', modelConfig?.mutations?.create);
     addServiceDirective(ctx.output.getMutationTypeName()!, 'update', modelConfig?.mutations?.update);
     addServiceDirective(ctx.output.getMutationTypeName()!, 'delete', modelConfig?.mutations?.delete);
     // @index queries
     if (ctx.metadata.has(indexKeyName)) {
       ctx.metadata.get<Set<string>>(indexKeyName)!.forEach(index => {
-        addServiceDirective(ctx.output.getQueryTypeName(), 'read', index.split(':')[1]);
+        addServiceDirective(ctx.output.getQueryTypeName(), 'list', index.split(':')[1]);
+        addServiceDirective(ctx.output.getQueryTypeName(), 'get', index.split(':')[1]);
       });
     }
     // @searchable
     if (searchableDirective) {
       const config = getSearchableConfig(searchableDirective, def.name.value);
-      addServiceDirective(ctx.output.getQueryTypeName(), 'read', config.queries.search);
+      addServiceDirective(ctx.output.getQueryTypeName(), 'search', config.queries.search);
     }
 
     const subscriptions = modelConfig?.subscriptions;
     if (subscriptions?.level === SubscriptionLevel.on) {
       const subscriptionArguments = acm
-        .getRolesPerOperation('read')
+        .getRolesPerOperation('listen')
```

Comment:
What if a user have get access on a field but not listen access? Do we set these fields to `null` in mutation response mapping? Because, currently our field level redaction support is not so great.